### ISSUE TITLE
table extension: Merged table columns

### ISSUE THREAD

**Issue by bep** [yuin/goldmark#69]
Tested with Goldmark 1.16.

```go
package main

import (
	"bytes"
	"fmt"
	"log"

	"github.com/yuin/goldmark"
	"github.com/yuin/goldmark/extension"
)

func main() {

	convert(`Foo|Bar
---|---
` + "`" + `Yoyo` + "`" + `|Dyne`)
}

func convert(src string) {

	markdown := goldmark.New(
		goldmark.WithExtensions(
			extension.Table,
		),
	)
	var buf bytes.Buffer
	err := markdown.Convert([]byte(src), &buf)
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(buf.String())
}

```

Produces 

```html
<table>
<thead>
<tr>
<th>Foo</th>
<th>Bar</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>Yoyo</code>|Dyne</td>
<td></td>
</tr>
</tbody>
</table>
```

The "try it" on https://commonmark.org/help/tutorial/02-emphasis.html renders it correclty, https://spec.commonmark.org/dingus/ renders nothing.

https://github.com/gohugoio/hugo/issues/6641

**Comment by frerich** [comment]
Thanks for the amazingly fast bugfix!

**Comment by moorereason** [comment]
For the record, it's fixed in commit 2aab93edb4513ad34d4ee59224fed6ad49de1370.  Commit e0171097bf26138f44befeef8ea9e5adf35f51a3 only added the failing test.
